genesyscloud/flow_logLevel/resource_genesyscloud_flow_logLevel.go

```diff
@@ -0,0 +1,153 @@
+utils function in the package.  This will keep the code manageable and easy to work through.
+*/
+// getAllFlowLogLevels retrieves all of the flow log levels via Terraform in the Genesys Cloud and is used for the exporter
+func getAllFlowLogLevels(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	ep := getFlowLogLevelProxy(clientConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	flowLogLevels, err := ep.getAllFlowLogLevels(ctx)
+	if err != nil {
+		return nil, diag.Errorf("Failed to get flow log levels: %v", err)
+	}
+
+	for _, flowLogLevel := range *flowLogLevels {
+		log.Printf("Dealing with flow log level id : %s", *flowLogLevel.Id)
+		resources[*flowLogLevel.Id] = &resourceExporter.ResourceMeta{Name: *flowLogLevel.Id}
+	}
+
+	return resources, nil
+}
+
+// createFlowLogLevel is used by the flow_logLevel resource to create Genesyscloud flow_logLevel
+func createFlowLogLevel(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	ep := getFlowLogLevelProxy(sdkConfig)
+	flowId := d.Get("flow_id").(string)
+	flowLogLevelRequest := getFlowLogLevelSettingsRequestFromResourceData(d)
+
+	flowLogLevel, err := ep.createFlowLogLevel(ctx, flowId, &flowLogLevelRequest)
+	if err != nil {
+		return diag.Errorf("Failed to create flow log level: %s", err)
+	}
+
+	d.SetId(*flowLogLevel.Id)
+	return readFlowLogLevel(ctx, d, meta)
+}
+
+// readFlowLogLevels is used by the flow_logLevel resource to read a flow log level from genesys cloud.
+func readFlowLogLevel(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	ep := getFlowLogLevelProxy(sdkConfig)
+	flowId := d.Get("flow_id").(string)
+
+	log.Printf("Reading readFlowLogLevel with flowId %s", flowId)
+	if flowId == "" {
+		log.Printf("flow log level with blank flowId %s", flowId)
+		return diag.Errorf("flowId not found")
+	}
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		flowSettingsResponse, respCode, err := ep.getFlowLogLevelById(ctx, flowId)
+		if err != nil {
+			log.Print(err)
+			if util.IsStatus404ByInt(respCode) {
+				return retry.NonRetryableError(fmt.Errorf("Failed to read flow log level %s: %s", flowId, err))
+			}
+			return retry.NonRetryableError(fmt.Errorf("Failed to read flow log level %s: %s", flowId, err))
+		}
+
+		flowLogLevel := flowSettingsResponse.LogLevelCharacteristics
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceFlowLoglevel())
+		resourcedata.SetNillableValue(d, "flow_log_level", flowLogLevel.Level)
+		resourcedata.SetNillableValueWithInterfaceArrayWithFunc(d, "flow_characteristics", flowLogLevel.Characteristics, flattenFlowCharacteristics)
+
+		log.Printf("Read flow log level %s", flowId)
+		checkState := cc.CheckState()
```

Comment:
Instead of getting the value returned from `cc.CheckState()` and printing it, you can do `return cc.CheckState()`